scalding-core/src/main/scala/com/twitter/scalding/mathematics/Matrix2.scala

```diff
@@ -359,6 +359,13 @@ case class Product[R, C, C2, V](left: Matrix2[R, C, V],
 
   implicit override val rowOrd: Ordering[R] = left.rowOrd
   implicit override val colOrd: Ordering[C2] = right.colOrd
+  implicit def withOrderedSerialization: Ordering[(R, C2)] = {
```

Comment:
Can we not repeat this code 3 times? What if we have this function in the companion object of `OrderedSerialization2` in this file:

https://github.com/twitter/scalding/blob/develop/scalding-serialization/src/main/scala/com/twitter/scalding/serialization/Serialization2.scala#L61

```scala
def maybeOrderedSerialization[A, B](implicit ordA: Ordering[A], ordB: Ordering[B]): Ordering[(A, B)] = ...
```